metals-docs/src/main/scala/docs/VersionedDocsModifier.scala

```diff
@@ -0,0 +1,25 @@
+package docs
+
+import mdoc.Reporter
+import mdoc.StringModifier
+import scala.meta.inputs.Input
+import scala.meta.internal.metals.BuildInfo
+import scala.meta.internal.semver.SemVer.Version
+
+class VersionedDocsModifier extends StringModifier {
+  val name = "since"
+
+  override def process(
+      version: String,
+      code: Input,
+      reporter: Reporter
+  ): String = {
+    val unreleased =
+      Version.fromString(version) >= Version.fromString(BuildInfo.metalsVersion)
+
+    if (unreleased) {
+      s"> 🚧 This documentation section describes a feature that is currently unreleased. The expected release version is $version\n"
+    } else ""
```

Comment:
This is meant to make sure we don't accidentally leave the warning visible when the corresponding version has been published.

Of course we could just remove the annotation instead, this is a safeguard.

Reply:
Alternatively, we can keep these and change the message into something like
```suggestion
    } else {
      "**Available since**: v0.8.4, released April 2020.\n"
    }
```
We can probably drill out the release date from git. Leaving it empty is also fine 👍

Reply:
Nit, I try to keep if/else branches consistent wrt to braces
```suggestion
    } else {
      ""
    }
```